src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5394,6 +5417,275 @@ def delete_deployment(self, deployment_id: UUID) -> None:
+                session=session,
+                artifact_visualization_id=visualization.artifact_visualization_id,
+                resource_id=visualization.resource_id,
+                resource_type=visualization.resource_type,
+            )
+            if visualization.display_order is not None:
+                self._assert_curated_visualization_display_order_unique(
+                    session=session,
+                    resource_id=visualization.resource_id,
+                    resource_type=visualization.resource_type,
+                    display_order=visualization.display_order,
+                )
+
+            schema = CuratedVisualizationSchema.from_request(visualization)
+
+            session.add(schema)
+            session.commit()
+            session.refresh(schema)
+
+            return schema.to_model(
+                include_metadata=True,
+                include_resources=True,
+            )
+
+    def get_curated_visualization(
+        self, visualization_id: UUID, hydrate: bool = True
+    ) -> CuratedVisualizationResponse:
+        """Fetch a curated visualization by ID.
+
+        Args:
+            visualization_id: The ID of the curated visualization to fetch.
+            hydrate: Flag deciding whether to hydrate the output model(s)
+                by including metadata fields in the response.
+
+        Returns:
+            The curated visualization with the given ID.
+        """
+        with Session(self.engine) as session:
+            schema: CuratedVisualizationSchema = self._get_schema_by_id(
+                resource_id=visualization_id,
+                schema_class=CuratedVisualizationSchema,
+                session=session,
+            )
+            return schema.to_model(
+                include_metadata=hydrate,
+                include_resources=hydrate,
+            )
+
+    def update_curated_visualization(
+        self,
+        visualization_id: UUID,
+        visualization_update: CuratedVisualizationUpdate,
+    ) -> CuratedVisualizationResponse:
+        """Update mutable fields on a curated visualization.
+
+        Args:
+            visualization_id: The ID of the curated visualization to update.
+            visualization_update: The update to apply to the curated visualization.
+
+        Returns:
+            The updated curated visualization.
+        """
+        with Session(self.engine) as session:
+            schema = self._get_schema_by_id(
+                resource_id=visualization_id,
+                schema_class=CuratedVisualizationSchema,
+                session=session,
+            )
+            update_fields = visualization_update.model_dump(exclude_unset=True)
+            if "display_order" in update_fields:
+                new_display_order = update_fields["display_order"]
+                if new_display_order is not None:
+                    self._assert_curated_visualization_display_order_unique(
```

Comment:
This is prone to race conditions if multiple requests to update display order happen at the same time. I'm okay if we merge this now because we're short on time, but the real solution would be a unique index on resource type, resource ID and display order.